content/doc/book/pipeline/pipeline-best-practices.adoc

```diff
@@ -16,28 +16,36 @@ endif::[]
 
 = Applying best practices to Pipelines
 
-This guide provides a small selection of best practices for pipelines and points out the most common mistakes. 
+This guide provides a small selection of best practices for pipelines and points out the most common mistakes.
 
 The goal is to point pipeline authors and maintainers towards patterns that result in better Pipeline execution and away from pitfalls they might otherwise not be aware of.
 This guide is not meant to be an exhaustive list of all possible Pipeline best practices but instead to provide a number of specific examples useful in tracking down common practices.
 Use it as a "do this" generally and not as an incredibly detailed "how-to".
 
-This guide is arranged by area, guideline, then listing specific examples. 
+This guide is arranged by area, guideline, then listing specific examples.
 
 == General
 
 === Making sure to use Groovy code in Pipelines as glue
 
 Use Groovy code to connect a set of actions rather than as the main functionality of your Pipeline.
-In other words, instead of relying on Pipeline functionality (Groovy or Pipeline steps) to drive the build process forward, use single steps (such as `sh`) to accomplish multiple parts of the build. 
+In other words, instead of relying on Pipeline functionality (Groovy or Pipeline steps) to drive the build process forward, use single steps (such as `sh`) to accomplish multiple parts of the build.
 Pipelines, as their complexity increases (the amount of Groovy code, number of steps used, etc.), require more resources (CPU, memory, storage) on the controller.
 Think of Pipeline as a tool to accomplish a build rather than the core of a build.
 
 Example: Using a single Maven build step to drive the build through its build/test/deploy process.
 
+=== Running shell scripts in Jenkins Pipeline
+
+Using a shell script within Jenkins Pipeline can help simplify builds, by combining multiple steps into a single stage. The shell script also allows users to add or update commands without having to modify each location separately.
```

Comment:
```suggestion
Using a shell script within Jenkins Pipeline can help simplify builds by combining multiple steps into a single stage. The shell script also allows users to add or update commands without having to modify each location separately.
```

Reply:
What does "location" refer to in line 40? Wondering if it would it be more precise to say:

... without having to modify each **step** separately.